app/frontend/helpers/cliParser/parser.tsx

```diff
@@ -172,21 +211,58 @@ const parseTxCerts = (txCertificates: any[]): _Certificate[] => {
   return txCertificates.map((certificate) => parseTxCert(certificate))
 }
 
-const parseTxWithdrawals = (withdrawals: Map<Buffer, number>): _Withdrawal[] => {
+const parseTxWithdrawals = (withdrawals: any): _Withdrawal[] => {
   if (!isWithdrawalsMap(withdrawals)) {
     throw Error(Errors.WithrawalsParseError)
   }
-  return Array.from(withdrawals).map(([address, coins]): _Withdrawal => ({address, coins}))
+  return Array.from(withdrawals).map(
+    ([address, coins]): _Withdrawal => ({address, coins: BigInt(coins)})
+  )
+}
+
+const parseFee = (fee: any): Lovelace => {
+  if (!isUint64(fee)) {
+    throw Error(Errors.FeeParseError)
+  }
+  return BigInt(fee)
+}
+
+const parseTtl = (ttl: any): BigInt | undefined => {
+  if (ttl && !isUint64(ttl)) {
+    throw Error(Errors.TTLParseError)
+  }
+  return ttl && BigInt(ttl)
+}
+
+const parseValidityIntervalStart = (validityIntervalStart: any): BigInt | undefined => {
+  if (validityIntervalStart && !isUint64(validityIntervalStart)) {
+    throw Error(Errors.ValidityIntervalStartParseError)
+  }
+  return validityIntervalStart && BigInt(validityIntervalStart)
+}
+
+const parseMetaDataHash = (metaDataHash: any): Buffer | undefined => {
+  if (metaDataHash && !Buffer.isBuffer(metaDataHash)) {
+    throw Error(Errors.MetaDataHashParseError)
+  }
+  return metaDataHash
 }
 
 const parseUnsignedTx = ([txBody, meta]: _UnsignedTxDecoded): _UnsignedTxParsed => {
+  if (txBody.get(TxBodyKeys.MINT)) {
```

Comment:
wouldn't it be a more resilient/simpler approach to serialize the tx again and compare if it matches byte-by-byte with the originally supplied tx?